org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/telemetry/SonarLintTelemetry.java

```diff
@@ -0,0 +1,172 @@
+        } catch (Exception e) {
+          // Silently ignore
+        }
+      }
+    }
+  }
+
+  public boolean enabled() {
+    return enabled;
+  }
+
+  public boolean optedIn() {
+    return enabled && this.telemetryEngine.enabled();
+  }
+
+  public void init() {
+    if ("true".equals(System.getProperty(DISABLE_PROPERTY_KEY))) {
+      this.enabled = false;
+      SonarLintLogger.get().info("Telemetry disabled by system property");
+      return;
+    }
+    try {
+      this.telemetryEngine = new Telemetry(getStorageFilePath(), PRODUCT, SonarLintUtils.getPluginVersion());
+      SonarLintCorePlugin.getAnalysisListenerManager().addListener(this);
+      this.scheduledJob = new TelemetryJob();
+      scheduledJob.schedule(TimeUnit.MINUTES.toMillis(1));
+      this.enabled = true;
+    } catch (Exception e) {
+      // Silently ignore
+      enabled = false;
+    }
+  }
+
+  private class TelemetryJob extends Job {
+
+    public TelemetryJob() {
+      super("SonarLint Telemetry");
+    }
+
+    protected IStatus run(IProgressMonitor monitor) {
+      schedule(TimeUnit.HOURS.toMillis(6));
+      if (enabled) {
+        try {
+          TelemetryClientConfig clientConfig = getTelemetryClientConfig();
+          telemetryEngine.getClient().tryUpload(clientConfig, isAnyProjectConnected());
+        } catch (Exception e) {
+          // Silently ignore
+        }
+      }
+      return Status.OK_STATUS;
+    }
+
+  }
+
+  public static TelemetryClientConfig getTelemetryClientConfig() {
+    TelemetryClientConfig.Builder clientConfigBuilder = new TelemetryClientConfig.Builder()
+      .userAgent("SonarLint");
+    IProxyService proxyService = SonarLintCorePlugin.getDefault().getProxyService();
+    IProxyData[] proxyDataForHost;
+    try {
+      proxyDataForHost = proxyService.select(new URL(Telemetry.TELEMETRY_ENDPOINT).toURI());
+    } catch (MalformedURLException | URISyntaxException e) {
+      // URL is a constant, should never occurs
+      throw new IllegalStateException(e);
+    }
+
+    for (IProxyData data : proxyDataForHost) {
+      if (data.getHost() != null) {
+        clientConfigBuilder.proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress(data.getHost(), data.getPort())));
+        if (data.isRequiresAuthentication()) {
+          clientConfigBuilder.proxyLogin(data.getUserId());
+          clientConfigBuilder.proxyPassword(data.getUserId());
+        }
+        break;
+      }
+    }
+    return clientConfigBuilder.build();
+  }
+
+  @Override
+  public void analysisCompleted(AnalysisEvent event) {
+    if (enabled) {
+      telemetryEngine.getDataCollection().analysisDone();
+    }
+  }
+
+  public void stop() {
+    SonarLintCorePlugin.getAnalysisListenerManager().removeListener(this);
+    if (scheduledJob != null) {
+      scheduledJob.cancel();
+      scheduledJob = null;
+    }
+    try {
+      if (telemetryEngine != null) {
+        telemetryEngine.save();
+      }
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1166)